dos/src/main.rs

```diff
@@ -173,100 +299,203 @@ fn main() {
                     "repair",
                     "serve_repair",
                     "rpc",
-                ])
-                .help("Interface to DoS"),
-        )
-        .arg(
-            Arg::with_name("data_size")
-                .long("data-size")
-                .takes_value(true)
-                .value_name("BYTES")
-                .help("Size of packet to DoS with"),
-        )
-        .arg(
-            Arg::with_name("data_type")
-                .long("data-type")
-                .takes_value(true)
-                .value_name("TYPE")
-                .possible_values(&[
+                ],
+        parse(try_from_str = mode_parser),
+        help="Interface to DoS")]
+    mode: Mode,
+
+    #[clap(
+        long = "data-size",
+        default_value = "128",
+        required_if_eq("data-type", "random"),
+        help = "Size of packet to DoS with, relevant only for data-type=random"
+    )]
+    data_size: usize,
+
+    #[clap(long="data-type",
+        possible_values=&[
```

Comment:
I'm surprised we still need this, although I haven't played with clap derive before.
Can we not just `rename_all = snake_case` or something? https://github.com/clap-rs/clap/blob/v3.1.6/examples/derive_ref/README.md#arg-enum-attributes

I assumed clap would handle the parsing for enums too.

Reply:
Also, what do you think about changing these to kebab-case? Snake is inconsistent with our CLI tools everywhere.
Will it break anything already out in the wild?

Reply:
I'm a bit confused here. If I use `rename_all=...`, arguments will become positional. So if I run `help` it will show 

```bash
ARGS:
     <MODE>   Interface to DoS [...]
```

If I instead specify long help gives me
```bash
--mode <MODE>    Interface to DoS  ...
```

I would expect that this `rename_all` option will create `long` for me automatically by following kebab-case rules, but it doesn't happen.


Regarding second comment -- I think I use `kebab-case` everywhere in the `long` attribute.

Reply:
> Also, what do you think about changing these to kebab-case? Snake is inconsistent with our CLI tools everywhere.

I was referring here to the `possible_values`, not the `long` definition

Reply:
I see.

I figured out how to simplify a bit code for `long` -- just write `long` without explicit value and this will lead to assigning proper long name automatically:
```rust
#[clap(rename_all = "kebab-case")]
struct A {
    #[clap(long)] // the same as if I wrote long = "param-1"
    param_1: type
}
```

But it seems that the same mechanism doesn't work for `enums` values so I have to specify default values, parser, etc.

Reply:
Upd: `ArgEnum` might be solution